packages/website/docs/examples/alert-block.md

```diff
@@ -0,0 +1,464 @@
+    "alert"
+  >;
+  editor: BlockNoteEditor<
+    DefaultBlockSchema & { alert: BlockSpec<"alert", typeof alertPropSchema> }
+  >;
+}) => {
+  const [type, setType] = useState(props.block.props.type);
+  const Icon = alertTypes[type].icon;
+
+  const alertStyles = {
+    display: "flex",
+    justifyContent: "center",
+    alignItems: "center",
+    flexGrow: 1,
+    borderRadius: "20px",
+    padding: "4px",
+  } as const;
+
+  const alertWrapperStyles = {
+    backgroundColor: alertTypes[type].color,
+    borderRadius: "16px",
+    display: "flex",
+    justifyContent: "center",
+    alignItems: "center",
+    marginRight: "0.5rem",
+    height: "32px",
+    width: "32px",
+    userSelect: "none",
+    cursor: "pointer",
+  } as const;
+
+  const inlineContentStyles = {
+    flexGrow: "1",
+  };
+
+  return (
+    <div className={"alert"} style={alertStyles}>
+      {/*Icon which opens a menu to choose the Alert type*/}
+      <Menu>
+        <Menu.Target>
+          <div
+            className={"alert-icon-wrapper"}
+            style={alertWrapperStyles}
+            contentEditable={false}>
+            <Icon className={"alert-icon"} size={32} />
+          </div>
+        </Menu.Target>
+        <Menu.Dropdown>
+          <Menu.Label>Alert Type</Menu.Label>
+          <Menu.Divider />
+          {Object.entries(alertTypes).map(([key, value]) => {
+            const ItemIcon = value.icon;
+
+            return (
+              <Menu.Item
+                key={key}
+                icon={<ItemIcon color={value.color} />}
+                onClick={() => setType(key as keyof typeof alertTypes)}>
+                {key.slice(0, 1).toUpperCase() + key.slice(1)}
+              </Menu.Item>
+            );
+          })}
+        </Menu.Dropdown>
+      </Menu>
+      {/*Rich text field for user to type in*/}
+      <InlineContent style={inlineContentStyles} />
+    </div>
+  );
+};
+
+// The Alert block itself
+export const AlertBlock = createReactBlockSpec({
+  type: "alert" as const,
+  propSchema: {
+    textAlignment: defaultProps.textAlignment,
+    textColor: defaultProps.textColor,
+    type: {
+      default: "warning",
+      values: ["warning", "error", "info", "success"],
+    },
+  } as const,
+  containsInlineContent: true,
+  render: Alert,
+});
+
+// Slash menu item to insert an Alert block
+export const insertAlert = {
+  name: "Alert",
+  execute: (editor) => {
+    const block = editor.getTextCursorPosition().block;
+    const blockIsEmpty = block.content.length === 0;
+
+    // Updates current block to an Alert if it's empty, otherwise inserts a new
+    // one below
+    if (blockIsEmpty) {
+      editor.updateBlock(block, { type: "alert" });
+    } else {
+      editor.insertBlocks(
+        [
+          {
+            type: "alert",
+          },
+        ],
+        editor.getTextCursorPosition().block,
+        "after"
+      );
+      editor.setTextCursorPosition(editor.getTextCursorPosition().nextBlock!);
```

Comment:
`insertBlocks` can insert multiple blocks though, so it gets pretty cumbersome if e.g. the inserted blocks don't have IDs (which is usually the case), since you have to retrieve all the blocks again after you insert them, and can't do it by ID. So I think it's actually quite a bit of work to implement properly, especially since it would make sense to do the same thing for `updateBlock`, `replaceBlocks`, etc